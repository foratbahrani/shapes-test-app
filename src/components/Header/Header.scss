.header {
	background-color: #333;
	color: #fff;
	width: 100%;
	height: 80px;
	line-height: 80px;
}

.header h1 {
	font-size: 2em;
	font-weight: 500;
	margin-block-start: 0;
	margin-block-end: 0;
	display: inline;

	@media (max-width: 468px) {
		font-size: 1em;
	}
}

.headerFlexBox {
	display: flex;
	justify-content: space-between;
	flex-direction: row;
}

.header .refreshButton {
	font-size: 1em;
	border: none;
	background-color: transparent;
	color: #fff;
	cursor: pointer;
	-webkit-appearance: none;
	-moz-appearance: none;
	appearance: none;

	@media (max-width: 468px) {
		font-size: 0.8em;
	}
}

.header .refreshButton:focus {
	outline: 0;
}

.navbar {
	display: inline;
	list-style-type: none;
	margin: 0;
	padding: 0;
	overflow: hidden;
}

.navbar > li {
	float: left;
	padding: 0 16px;

	@media (max-width: 468px) {
		padding: 0 8px;
		font-size: 0.8em;
	}
}

.navbar > li > a {
	color: #777;
}

.navbar > li > a:hover {
	color: #aaa;
}

.navbar > li .active {
	color: white;
	text-decoration: underline;
}
