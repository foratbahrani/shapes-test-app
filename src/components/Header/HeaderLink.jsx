import React from 'react';
import { Link, withRouter } from 'react-router-dom';

export const HeaderLink = ({ to, children, history }) => {
	const { pathname } = history.location;
	const getClassName = () => (pathname.includes(to) ? 'active' : '');

	return (
		<Link to={to} className={getClassName()}>
			{children}
		</Link>
	);
};

export default withRouter(HeaderLink);
