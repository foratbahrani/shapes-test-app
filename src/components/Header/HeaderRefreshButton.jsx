import React from 'react';
import { connect } from 'react-redux';
import { fetchShapesStart } from '../../store/shapes/shapes.actions';

export const HeaderRefreshButton = ({ refreshShapes }) => {
	return (
		<button className='refreshButton shrink' id='button' onClick={refreshShapes}>
			Refresh
		</button>
	);
};

const mapDispatchToProps = {
	refreshShapes: fetchShapesStart,
};

export default connect(null, mapDispatchToProps)(HeaderRefreshButton);
