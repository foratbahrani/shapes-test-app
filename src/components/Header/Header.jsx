import React from 'react';
import Container from '../../containers/Container/Container';
import './Header.scss';
import HeaderLink from './HeaderLink';
import HeaderRefreshButton from './HeaderRefreshButton';

const Header = () => {
	return (
		<div className='header'>
			<Container size='lg'>
				<div className='headerFlexBox'>
					<ul className='navbar'>
						<li>
							<h1>SHAPES</h1>
						</li>
						<li>
							<HeaderLink to='/redux-saga'>ReduxSaga</HeaderLink>
						</li>
						<li>
							<HeaderLink to='/mvvm'>MVVM</HeaderLink>
						</li>
					</ul>

					<HeaderRefreshButton />
				</div>
			</Container>
		</div>
	);
};

export default Header;
