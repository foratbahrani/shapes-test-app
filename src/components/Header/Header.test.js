import { shallow } from 'enzyme';
import React from 'react';
import Header from './Header';

it('matches snapshot', () => {
	expect(shallow(<Header />)).toMatchSnapshot();
});
