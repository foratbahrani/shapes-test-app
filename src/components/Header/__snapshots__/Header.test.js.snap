// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="header"
>
  <Container
    size="lg"
  >
    <div
      className="headerFlexBox"
    >
      <ul
        className="navbar"
      >
        <li>
          <h1>
            SHAPES
          </h1>
        </li>
        <li>
          <withRouter(HeaderLink)
            to="/redux-saga"
          >
            ReduxSaga
          </withRouter(HeaderLink)>
        </li>
        <li>
          <withRouter(HeaderLink)
            to="/mvvm"
          >
            MVVM
          </withRouter(HeaderLink)>
        </li>
      </ul>
      <Connect(HeaderRefreshButton) />
    </div>
  </Container>
</div>
`;
